stackrox/stackrox
# Delete fixture from Configuration Management dashboard integration test

## Description

### Problem

logimbue-data.json file always contains at least one exception even when tests pass, which slows down investigation of integration test failures.

Tests added in https://github.com/stackrox/rox/pull/2929 and most recently edited in https://github.com/stackrox/rox/pull/9371

### Analysis

```json
"url": "https://central-lb/main/configmanagement/subjects/MDZkMWMzYjktOGQ5Ny00OTZkLTg0ZGItMGM1Nzg5MTBiNzFi%3Ac3lzdGVtOm1hc3RlcnM%3D"
```

```json
          "type": "TypeError",
          "value": "Cannot read properties of undefined (reading 'subject')",
```

```json
      "lastAction": { "type": "serverError/RECORD_SUCCESS" },
      "state": {
        "router": {
          "location": {
            "pathname": "/main/configmanagement/subjects/MDZkMWMzYjktOGQ5Ny00OTZkLTg0ZGItMGM1Nzg5MTBiNzFi%3Ac3lzdGVtOm1hc3RlcnM%3D",
            "search": "",
            "hash": "",
            "key": "y2nbu4",
            "query": {}
          },
          "action": "PUSH"
        },
```

Only one test related to Users & Groups (also known as subjects) has a click in svg.

It has **an unneeded** fixture for /api/graphql?opname=usersWithClusterAdminRoles

It does not have corresponding fixtures
* for /api/graphql?opname=getSubjectName
* for /api/graphql?opname=getSubject

```json
        {
          "timestamp": 1653326296.904,
          "category": "ui.click",
          "message": "div \u003e div.rv-xy-plot. \u003e svg \u003e g \u003e rect"
        },
        {
          "timestamp": 1653326296.904,
          "category": "navigation",
          "data": {
            "to": "/main/configmanagement/subjects/MDZkMWMzYjktOGQ5Ny00OTZkLTg0ZGItMGM1Nzg5MTBiNzFi%3Ac3lzdGVtOm1hc3RlcnM%3D",
            "from": "/main/configmanagement"
          }
        },

        {
          "timestamp": 1653326297.026,
          "type": "http",
          "category": "xhr",
          "data": {
            "method": "POST",
            "url": "/api/graphql?opname=getSubject",
            "status_code": 200
          }
        },
        {
          "timestamp": 1653326297.026,
          "category": "redux-action",
          "message": "serverError/RECORD_SUCCESS"
        },
        {
          "timestamp": 1653326297.034,
          "category": "sentry",
          "message": "Error: subject not found: system:masters: not found",
          "event_id": "d60cc86b578f4a728759069366b26fc4",
          "level": "error"
        },
        {
          "timestamp": 1653326297.049,
          "message": "TypeError: Cannot read properties of undefined (reading 'subject')",
          "level": "error",
          "category": "console"
        }
```

### Solution

1. Delete cypress/fixtures/subjects

2. Edit cypress/integration/configmanagement/dashboard.test.js
    * Delete the unneeded fixture from `'clicking a specific user in the "Users with most Cluster Admin Roles" widget should take you to a single subject page'` test
    * Replace single with list urls in 2 assertions

### Residue

1. Instead of page crash, render `error` from GraphQL query: https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/ConfigManagement/Entity/Subject.js#L86-L88

## Checklist

## Testing Performed

### manual testing

1. `yarn start` in ui

    /main/configmanagement/bogusId to see exception when GraphQL query has `subject: null` and `errors`
    ![ui-error](https://user-images.githubusercontent.com/11862657/172264840-4530e444-388b-42bd-bc45-5cd1d1ec3fb7.png)

### integration testing

1. `yarn cypress-open` in ui/apps/platform

    * configmanagement/dashboard.test.js

        With fixture
        <img width="1284" alt="cypress-with-fixture" src="https://user-images.githubusercontent.com/11862657/172264807-2235f2ad-4420-4659-a450-ecaccb8b0100.png">

        Without fixture
        <img width="1284" alt="cypress-without-fixture" src="https://user-images.githubusercontent.com/11862657/172264779-9f04bbbb-3ed2-49df-b89a-37746a0a2686.png">

    * configmanagement/usersandgroups.test.js
